functions/v2/typed/googlechatbot/main.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# [START functions_typed_googlechatbot]
+from dataclasses import dataclass
+from typing import Any, Dict, List, Union
+
+import functions_framework
+
+
+@dataclass
+class ChatRequest:
+    message: Dict[str, Union[str, Dict[str, str], List[Dict[str, str]]]]
+
+    # Required to deserialize the request
+    @staticmethod
+    def from_dict(req: dict) -> "ChatRequest":
```

Comment:
if forward declarations isn't an easy option, and if this is only used within tests, would it be easier to port this logic out onto the tests?

Reply:
Got it. Not the most preferred way but I won't make this a blocker for you.